packages/alfa-style/src/property/contain.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  Keyword,
+  Token,
+  Tuple,
+  type Parser as CSSParser,
+} from "@siteimprove/alfa-css";
+import { Parser } from "@siteimprove/alfa-parser";
+import { Slice } from "@siteimprove/alfa-slice";
+import { Err, Result } from "@siteimprove/alfa-result";
+
+import { Longhand } from "../longhand.js";
+
+const { either, map } = Parser;
+
+type Specified = Specified.Single | Specified.Multiple;
+type Computed = Specified;
+
+namespace Specified {
+  export type Single = Tuple<
+    [Keyword<"none"> | Keyword<"strict"> | Keyword<"content">]
+  >;
+  export type Multiple = Tuple<
+    [
+      Keyword<"size"> | Keyword<"inline-size">,
+      Keyword<"layout">,
+      Keyword<"style">,
+      Keyword<"paint">,
+    ]
+  >;
+}
+
+const parseSingle: CSSParser<Specified.Single> = map(
+  Keyword.parse("none", "strict", "content"),
+  (keyword) => Tuple.of(keyword),
+);
+const parseMultiple: CSSParser<Specified.Multiple> = (input) => {
+  let size: Keyword<"size"> | Keyword<"inline-size"> | undefined;
+  let layout: Keyword<"layout"> | undefined;
+  let style: Keyword<"style"> | undefined;
+  let paint: Keyword<"paint"> | undefined;
+
+  while (true) {
+    for ([input] of Token.parseWhitespace(input)) {
+    }
```

Comment:
```suggestion
    for (const [remainder] of Token.parseWhitespace(input)) {
      input = remainder;
    }
```
I think that needs to update the input 🤔

Reply:
It is being updated because of `for([input] ...`, but I agree that's probably a little too hidden/obscure. I'll change it 👍🏼